### PR TITLE
Cleanup after getting HR off the island

### LINKED ISSUE #1713: Remove `uiowa_bar` from dependencies
Blocked until uids_base is adopted by all SiteNow service sites.

- [x] https://github.com/uiowa/uiowa/issues/1939 - turn on uiowa_search, migrate config

### LINKED ISSUE #1907: Remove `uiowa_footer` from dependencies
Blocked by #1824

### LINKED ISSUE #2629: Convert HR to uids_base
Originally 12-25 hours estimate for developer work. Does not include possible student work in replacing fontawesome icons with brand-approved versions and the scope has increased with recent work related to events, uiowa_bar, etc.

- [x] #3707
- [x] #3708
- [x] #2611 
- [x] https://github.com/uiowa/uiowa/issues/1606
- [x] Remove https://github.com/uiowa/uiowa/pull/2711/commits/9fe9bd5ea9b5b959116463c2136724755e015572 per https://github.com/uiowa/uiowa/pull/2711
- [x] Remove `config_ignore.settings` from `hr` site split.

## Related clean-up tasks
- [x] #1907 
- [x] #1713 
- [x] Remove https://github.com/uiowa/uiowa/blob/main/docroot/modules/custom/sitenow_people/sitenow_people.module#L269-L287 per https://github.com/uiowa/uiowa/pull/2906
- [x] #3719

### LINKED ISSUE #2630: Remove custom Composer install path for drupal-custom-module packages
### Problem
Currently, we install drupal-custom-modules to the custom path `docroot/modules/uiowa` because they are not contrib but can't go in custom otherwise they'd be tracked in VCS. However, this is a really odd location and could cause confusion if we use a non-UI custom module in the future.

### Possible solutions
1. ~~Remove the custom install path and treat anything not in this repository as contributed. This would move uiowa_auth to `docroot/modules/contrib` which would require a cache rebuild.~~

2. Copy/paste uiowa_auth into this repository in `docroot/modules/custom` and ~~then make use of GitHub release file uploads for extracting and sharing that module, _if necessary_~~.  Update the composer.json file to remove the custom install path to `docroot/modules/uiowa`.

~~This is blocked by #1907 and #1713 which are blocked by #2629.~~

### LINKED ISSUE #3719: Remove `uiowa_bootstrap` and `hr` themes
After #3709 is merged, I believe we have no more use for the `uiowa_bootstrap` or `hr` themes, so they can be removed.

### PR DESCRIPTION
**Pull request by pyrello**
* Resolves #1907 
* Resolves #1713 
* Resolves #3719 
* ~Changes location of `uiowa_*` modules as outlined in #2629, but does not move `uiowa_auth` into this project.~ - Holding off on this until we can get `uiowa_auth` moved into the main project.
* Resolves #2629

# How to test

* Code review of changed files and skimming the deletions to make sure they make sense.
* `composer install`
* `blt ds --site hr.uiowa.edu`
* Ensure sync happens without error.
* `drush @hr.local uli`
* Spot check HR site locally to make sure there are no issues.
  * You can save card paragraphs without error
  * Article full images are displaying properly
  * Footer contact info is still present.
  * You can hide/show titles/sidebar on page content

### CODE REVIEW

File: docroot/profiles/custom/sitenow/sitenow.profile
Comment (pyrello) [inline docroot/profiles/custom/sitenow/sitenow.profile:198]: @joewhitsitt Can you validate this change?

File: docroot/profiles/custom/sitenow/sitenow.profile
Comment (pyrello) [inline docroot/profiles/custom/sitenow/sitenow.profile]: @joewhitsitt or @bspeare Can you weigh in?

File: docroot/themes/custom/uids_base/uids_base.theme
Comment (pyrello) [inline docroot/themes/custom/uids_base/uids_base.theme]: @joewhitsitt or @bspeare Can you weigh in?

File: docroot/profiles/custom/sitenow/sitenow.profile
Comment (joewhitsitt) [reply docroot/profiles/custom/sitenow/sitenow.profile:198]: yes this can be removed

**Comment by pyrello** [comment]
I can't actually test this against HR until the previous updates make it to prod.

**Comment by joewhitsitt** [comment]
>I can't actually test this against HR until the previous updates make it to prod.

deployment isn't done but HR has received the changes

File: docroot/profiles/custom/sitenow/sitenow.profile
Comment (joewhitsitt) [reply docroot/profiles/custom/sitenow/sitenow.profile]: Yes
```
// @todo Can this be removed?
$variables["title_attributes"]['class'][] = 'element-invisible';
```
 and the `case 'field_block:node:page:title':` can be removed.

This code block/element-invisible needs to remain for uids_base/page_title_block

File: docroot/themes/custom/uids_base/uids_base.theme
Comment (joewhitsitt) [reply docroot/themes/custom/uids_base/uids_base.theme]: i don't think this one can be removed. this is setting a publish option that sitenow.profile preprocess block then references and sets element-invisible for uids_base page_title_block

Comment (pyrello) [reply docroot/themes/custom/uids_base/uids_base.theme]: I should have included the line above in the snippet here, it is this:
```php
// For uiowa_bootstrap/classy default.
```

Comment (joewhitsitt) [reply docroot/themes/custom/uids_base/uids_base.theme]: i think that is an old comment because work was later done to get this working for page_title_block.

https://github.com/uiowa/uiowa/commit/9d1b0c5f1d9855c2230bff6293715c11e055edbd

File: docroot/profiles/custom/sitenow/sitenow.profile
Comment (pyrello) [reply docroot/profiles/custom/sitenow/sitenow.profile]: When I removed `case 'field_block:node:page:title':`, it cause the title to show up on the homepage, so I restored it. The other part was able to be removed.

**Comment by pyrello** [comment]
Because of moving the `uiowa_auth` module, this update should be scheduled for off-hours. And/or we should trigger `drush cr` for all sites just after deploying it.

**Comment by richardbporter** [comment]
Shouldn't running `composer install` be part of this? I pulled and ran that but uiowa_bar and uiowa_footer still here. Maybe people just need to delete the uiowa directory manually?

**Comment by pyrello** [comment]
> Shouldn't running `composer install` be part of this? I pulled and ran that but uiowa_bar and uiowa_footer still here. Maybe people just need to delete the uiowa directory manually?

Good call. Yes!

**Comment by richardbporter** [comment]
> Because of moving the `uiowa_auth` module, this update should be scheduled for off-hours. And/or we should trigger `drush cr` for all sites just after deploying it.

Yea I'm seeing a fatal error for all pages and even `drush cr`. Found this: https://www.drupal.org/project/drupal/issues/2160091 so `drush cr` might fix this in #4130. Otherwise we'd need an update hook to run it. Off-hours seems like a good option.

**Comment by richardbporter** [comment]
Another option would be to hold off on moving uiowa_auth here. Otherwise we have to do another off-hours deployment in https://github.com/uiowa/uiowa/issues/2630 when it moves to custom.

**Comment by pyrello** [comment]
I have reverted the changes related to moving `uiowa_auth` in favor of moving it all in one go in #2630 to minimize off-hours deployments.

Also, to be clear, above I meant running `blt ume cr` at the time of deployment. I don't think it is worth writing an update hook because there are sites that will still WSOD for an hour before the update hook runs.

**Comment by richardbporter** [comment]
> I have reverted the changes related to moving `uiowa_auth` in favor of moving it all in one go in #2630 to minimize off-hours deployments.
> 
> Also, to be clear, above I meant running `blt ume cr` at the time of deployment. I don't think it is worth writing an update hook because there are sites that will still WSOD for an hour before the update hook runs.

Yea thats a good idea. `ume` loops sites sequentially so they would WSOD still but it would take less time than running everything for each site. Maybe we could change our deployments up to loop sites twice (or make it optional) in the future: once to clear caches and another to updb and cim.

I tested this a bit differently because I was thinking we don't always catch these WSODs. I synced a site on main, hit site in in my browser, checked out this branch, ran `composer install` and deleted the `docroot/modules/uiowa` directory because that is not getting removed locally and it would not have existed in prod when the artifact was built. Hitting the site again then threw:
```
The website encountered an unexpected error. Please try again later.
AssertionError: The file specified by the given app root, relative path and file name (/var/www/uiowa/docroot/modules/uiowa/uiowa_auth/uiowa_auth.info.yml) do not exist. in assert() (line 67 of core/lib/Drupal/Core/Extension/Extension.php).
```

I couldn't get `drush cr` to even run -`drush updb` was the only thing that would run which is why I suggested putting one in sitenow.install with `drupal_flush_all_caches();` in it.

I will test this again now.

**Comment by richardbporter** [comment]
NOICE. 👍🏻 Gonna leave approval up to @joewhitsitt because he committed a few things here and I'm not sure if he has any other changes in mind.

**Comment by pyrello** [comment]
@richardbporter Thanks for the additional information and the additional testing.

**Comment by joewhitsitt** [comment]
testing one more time

**Review by joewhitsitt — APPROVED** [review]
End of an era

**Comment by joewhitsitt** [comment]
syncing a fresh copy of hr db to stage

**Pull request merged**
